spec/compiler/codegen/arithmetics_spec.cr

```diff
@@ -246,15 +246,50 @@ describe "Code gen: arithmetic primitives" do
       {% end %}
 
       {% for float_type in [Float32, Float64] %}
-        {% if type != UInt128 || float_type != Float32 %}
-          # skip for type == UInt128 && float_type == Float32
-          # since Float32::MAX < UInt128::MAX
-          it "raises overflow if greater than {{type}}::MAX (from {{float_type}})" do
+        it "raises overflow if greater than {{type}}::MAX (from {{float_type}})" do
+          {% if type == UInt128 && float_type == Float32 %}
+            # since `Float32::MAX < UInt128::MAX`, we have to ensure that the
+            # former does not overflow while positive infinity does
+            run(%(
+              require "prelude"
+
+              begin
+                Float32::INFINITY.to_u128
+                0
+              rescue OverflowError
+                begin
+                  Float32::MAX.to_u128
+                  1
+                rescue OverflowError
+                  2
+                end
+              end
+            )).to_i.should eq(1)
+          {% else %}
             run(%(
               require "prelude"
 
               v = {{float_type}}.new({{type}}::MAX) * {{float_type}}.new(1.5)
 
+              begin
+                v.{{method}}
+                0
+              rescue OverflowError
+                1
+              end
+            )).to_i.should eq(1)
+          {% end %}
+        end
+
+        {% if [Int64, Int128, UInt64].includes?(type) ||
+                (type == UInt128 && float_type == Float64) ||
+                ([Int32, UInt32].includes?(type) && float_type == Float32) %}
```

Comment:
This spec already has a very high complexity. This expression alone multiplies that by a lot. 

I fully agree to https://github.com/crystal-lang/crystal/pull/11229#issuecomment-922280101 that spec examples should be simple and reduce indirection to a minimum.
Can we find an easier way to express this?

Reply:
There is:

```crystal
it "raises overflow if equal to Int::MAX", tags: "primitives" do
  expect_raises(OverflowError) { Float32.new!(Int32::MAX).to_i32 }
  expect_raises(OverflowError) { Float32.new!(UInt32::MAX).to_u32 }
  expect_raises(OverflowError) { Float32.new!(Int64::MAX).to_i64 }
  expect_raises(OverflowError) { Float32.new!(UInt64::MAX).to_u64 }
  expect_raises(OverflowError) { Float32.new!(Int128::MAX).to_i128 }

  expect_raises(OverflowError) { Float64.new!(Int64::MAX).to_i64 }
  expect_raises(OverflowError) { Float64.new!(UInt64::MAX).to_u64 }
  expect_raises(OverflowError) { Float64.new!(Int128::MAX).to_i128 }
  expect_raises(OverflowError) { Float64.new!(UInt128::MAX).to_u128 }
end
```

#11233 removes arithmetics_spec.cr entirely, so I think this is fine.